substrate/client/chain-spec/src/chain_spec.rs

```diff
@@ -448,23 +451,29 @@ impl<G, E> ChainSpecBuilder<G, E> {
 		ChainSpec {
 			client_spec,
 			genesis: GenesisSource::GenesisBuilderApi(self.genesis_build_action, self.code.into()),
+			_host_functions: Default::default(),
 		}
 	}
 }
 
 /// A configuration of a chain. Can be used to build a genesis block.
-pub struct ChainSpec<G, E = NoExtension> {
+pub struct ChainSpec<G, E = NoExtension, HF = sp_io::SubstrateHostFunctions> {
```

Comment:
```suggestion
pub struct ChainSpec<G, E = NoExtension, ExtHF = ()> {
```

The "core" `SubstrateHostFunctions` IMO should always be present, and thus these are just the custom Extensions provided by the user. Core HF are injected below
Maybe add a couple of comments about what are these generic params